lib/api/response_models/deployment_response.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:client_common/api/response_models/api_response.dart';
+
+enum DeploymentStatus { created, pending, failure, success }
```

Comment:
Pending is not clear enough...

You should replace it by `waitingForBuild` and `waitingForAppReady`
```suggestion
enum DeploymentStatus { created, waitingForBuild, waitingForAppReady, failure, success }
```